atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/pue/CassandraImitatingConsensusForgettingStore.java

```diff
@@ -0,0 +1,166 @@
+public class CassandraImitatingConsensusForgettingStore implements ConsensusForgettingStore {
+    private final double probabilityOfFailure;
+    private final ThreadLocalRandom random = ThreadLocalRandom.current();
+    private final int quorum = 3;
+    private final AtomicLong timestamps = new AtomicLong(0);
+    private final List<Map<Cell, BytesAndTimestamp>> nodes = IntStream.range(0, 5)
+            .mapToObj(_ignore -> new HashMap<Cell, BytesAndTimestamp>())
+            .collect(Collectors.toList());
+
+    public CassandraImitatingConsensusForgettingStore(double probabilityOfFailure) {
+        this.probabilityOfFailure = probabilityOfFailure;
+    }
+
+    @Override
+    public void putUnlessExists(Cell cell, byte[] value) throws KeyAlreadyExistsException {
+        shuffleNodes();
+        for (int i = 0; i < quorum; i++) {
+            if (nodes.get(i).get(cell) != null) {
+                throw new KeyAlreadyExistsException("nice try");
+            }
+        }
+
+        writeToQuorum(cell, value);
+    }
+
+    @Override
+    public void putUnlessExists(Map<Cell, byte[]> values) throws KeyAlreadyExistsException {
+        values.forEach(this::putUnlessExists);
+    }
+
+    @Override
+    public void checkAndTouch(Cell cell, byte[] value) throws CheckAndSetException {
+        shuffleNodes();
+        ListenableFuture<Optional<byte[]>> read = get(cell);
+        if (!Arrays.equals(AtlasFutures.getDone(read).get(), value)) {
+            throw new CheckAndSetException("not happening");
+        }
+
+        writeToQuorum(cell, value);
+    }
+
+    @Override
+    public ListenableFuture<Optional<byte[]>> get(Cell cell) {
+        shuffleNodes();
+        Optional<BytesAndTimestamp> latest = IntStream.range(0, quorum)
+                .mapToObj(nodes::get)
+                .map(node -> node.get(cell))
+                .filter(Objects::nonNull)
+                .max(Comparator.comparing(BytesAndTimestamp::timestamp));
+        if (latest.isEmpty()) {
+            return Futures.immediateFuture(Optional.empty());
+        }
+        BytesAndTimestamp latestPresent = latest.get();
+        for (int i = 0; i < quorum; i++) {
+            nodes.get(i).put(cell, latestPresent);
+        }
+        return Futures.immediateFuture(latest.map(BytesAndTimestamp::bytes));
+    }
+
+    @Override
+    public ListenableFuture<Map<Cell, byte[]>> getMultiple(Iterable<Cell> cells) {
+        // too much effort to implement general case since it won't be used
+        Cell onlyCell = Iterables.getOnlyElement(cells);
+        return Futures.transform(
+                get(onlyCell),
+                maybeValue -> maybeValue
+                        .map(value -> ImmutableMap.of(onlyCell, value))
+                        .orElseGet(ImmutableMap::of),
+                MoreExecutors.directExecutor());
+    }
+
+    @Override
+    public void put(Cell cell, byte[] value) {
+        shuffleNodes();
+        for (int i = 0; i < quorum; i++) {
```

Comment:
seems like we want a `maybeFail` before beginning the loop - no reason that it always succeeds on the first node (I realise this may not matter in the current test driver, but it makes the model weird)